frontend/components/Corpus.js

```diff
@@ -0,0 +1,147 @@
+import * as PropTypes from "prop-types";
+// import STYLES from "./Corpus.module.scss";
+import {getCookie} from "../common";
+import {Modal} from "react-bootstrap";
+
+const Corpus = ({id}) => {
+
+    const [corpusData, setCorpusData] = useState({});
+    const [allDocData, setAllDocData] = useState([]);
+    const [containsDoc, setContainsDoc] = useState({});
+    const [loading, setLoading] = useState(true);
+    const [showModal, setShowModal] = useState(false);
+    const handleOpenModal = () => setShowModal(true);
+    const handleCloseModal = () => setShowModal(false);
+
+    useEffect(() => {
+        fetch(`/api/corpus/${id}`)
+            .then(response => response.json())
+            .then(data => {
+                setCorpusData(data);
+                fetch("/api/all_documents")
+                    .then(response => response.json())
+                    .then(docsData => {
+                        setAllDocData(docsData);
+                        docsData.map((doc) => setContainsDoc((values) => ({
+                            ...values,
+                            [doc.id]: data.documents.includes(doc.id)
+                        })));
+                        setLoading(false);
+                    });
+            });
+
+    }, []);
+
+    const handleCheckBoxChange = (event) => {
+        setContainsDoc((values) => ({
+            ...values,
+            [event.target.id]: !containsDoc[event.target.id]
+        }));
+    };
+
+    const updateDocs = (event) => {
+        event.preventDefault();
+        handleCloseModal();
+        const docList = Object.keys(containsDoc).filter((id) => {
+            return containsDoc[id];
+        });
+        const csrftoken = getCookie("csrftoken");
+        const requestOptions = {
+            method: "POST",
+            headers: {
+                "Content-Type": "application/json",
+                "X-CSRFToken": csrftoken
+            },
+            body: JSON.stringify({
+                id: corpusData.id,
+                documents: docList
+            })
+        };
+        fetch("/api/update_corpus_docs", requestOptions)
+            .then(response => response.json())
+            .then(data => {
+                setCorpusData(data);
+            });
+    };
+
+    const docsList = () => {
+        return (
+            <>
+                <h5>Documents:</h5>
+                {
+                    corpusData.documents.length
+                        ? <ul>
+                            {allDocData.map((doc, i) => {
+                                if (corpusData.documents.includes(doc.id)) {
+                                    return (
+                                        <li key={i}>
+                                            <i>{doc.title}</i>, by {doc.author} {doc.year &&
+                                                `(${doc.year})`}
+                                        </li>
+                                    );
+                                }
```

Comment:
I'm assuming that this map is working totally fine, but it looks like we're not supplying an explicit `return` value in the hypothetical `else` case, which I assume means this array is defaulting to supplying `undefined` in place of the non-targeted documents. This is working, but just to be more explicit, I think we could `.filter` out the `allDocData` and then map. That is: use `filter` to create the data structure we want and then `map` to convert that data structure to the DOM elements. That lets each array method handle its sole purpose rather than having `map` implicitly handling both. We could probably even pull up a new `const` that performs that `filter` and use it here and elsewhere.